drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -181,17 +181,19 @@ int intel_adsp_hda_dma_host_reload(const struct device *dev, uint32_t channel,
 #endif
 	switch (cfg->direction) {
 	case HOST_TO_MEMORY:
-		uint32_t rp = *DGBRP(cfg->base, cfg->regblock_size, channel);
-		uint32_t next_rp = (rp + INTEL_HDA_MIN_FPI_INCREMENT_FOR_INTERRUPT) %
+		; /* Only statements can be labeled in C, a declaration is not valid */
+		const uint32_t rp = *DGBRP(cfg->base, cfg->regblock_size, channel);
+		const uint32_t next_rp = (rp + INTEL_HDA_MIN_FPI_INCREMENT_FOR_INTERRUPT) %
 			intel_adsp_hda_get_buffer_size(cfg->base, cfg->regblock_size, channel);
 
 		intel_adsp_hda_set_buffer_segment_ptr(cfg->base, cfg->regblock_size,
 						      channel, next_rp);
 		intel_adsp_hda_enable_buffer_interrupt(cfg->base, cfg->regblock_size, channel);
 		break;
 	case MEMORY_TO_HOST:
-		uint32_t wp = *DGBWP(cfg->base, cfg->regblock_size, channel);
-		uint32_t next_wp = (wp + INTEL_HDA_MIN_FPI_INCREMENT_FOR_INTERRUPT) %
+		;
+		const uint32_t wp = *DGBWP(cfg->base, cfg->regblock_size, channel);
+		const uint32_t next_wp = (wp + INTEL_HDA_MIN_FPI_INCREMENT_FOR_INTERRUPT) %
```

Comment:
Right, so mixing code with declaration is not all roses ;)

While recognizing that this is pickled from upstream, I would have addressed this issue differently.
```diff
diff --git a/drivers/dma/dma_intel_adsp_hda.c b/drivers/dma/dma_intel_adsp_hda.c
index 14768e7531a5..2d62ee1c38b0 100644
--- a/drivers/dma/dma_intel_adsp_hda.c
+++ b/drivers/dma/dma_intel_adsp_hda.c
@@ -187,6 +187,7 @@ int intel_adsp_hda_dma_host_reload(const struct device *dev, uint32_t channel,
 #endif
 	switch (cfg->direction) {
 	case HOST_TO_MEMORY:
+	{
 		uint32_t rp = *DGBRP(cfg->base, cfg->regblock_size, channel);
 		uint32_t next_rp = (rp + INTEL_HDA_MIN_FPI_INCREMENT_FOR_INTERRUPT) %
 			intel_adsp_hda_get_buffer_size(cfg->base, cfg->regblock_size, channel);
@@ -195,7 +196,9 @@ int intel_adsp_hda_dma_host_reload(const struct device *dev, uint32_t channel,
 						      channel, next_rp);
 		intel_adsp_hda_enable_buffer_interrupt(cfg->base, cfg->regblock_size, channel);
 		break;
+	}
 	case MEMORY_TO_HOST:
+	{
 		uint32_t wp = *DGBWP(cfg->base, cfg->regblock_size, channel);
 		uint32_t next_wp = (wp + INTEL_HDA_MIN_FPI_INCREMENT_FOR_INTERRUPT) %
 			intel_adsp_hda_get_buffer_size(cfg->base, cfg->regblock_size, channel);
@@ -204,6 +207,7 @@ int intel_adsp_hda_dma_host_reload(const struct device *dev, uint32_t channel,
 						      channel, next_wp);
 		intel_adsp_hda_enable_buffer_interrupt(cfg->base, cfg->regblock_size, channel);
 		break;
+	}
 	default:
 		break;
 	}
```

Reply:
> Right, so mixing code with declaration is not all roses ;)

Someone forgot to update the formal language grammar in 1999. sparse seems to be the only one which cares.

BTW sparse is also unmaintained, can't do exit codes and has infinite loops:
- https://github.com/zephyrproject-rtos/zephyr/issues/63417

> I would have addressed this issue differently.

Feel free to send a patch upstream. If anyone cares enough we can then cherry-pick it on top of this one.

Reply:
Let's handle this in upstream, I can pick this one for the stable series as this is already upstream.